WildMeOrg/codex-frontend
# Dex 739 new block and merge notification implementation

Implements the last missing pieces of https://wildme.atlassian.net/browse/DEX-739.

Back end has made the eta as well as the other individuals' (although, for MVP, we're just hard-coding it as a pairwise merge, so more accurately individual's rather than individuals') name and guid available for the individual_merge_request notification type.

They have also made an individual_merge_blocked notification type available.

With these available, I have been able to flesh out the notification, notificationPane, and notificaiton dialogs of both individual_merge_blocked and individual_merge_request:

# individual_merge_request:
 - Notification:
![Screen Shot 2022-05-20 at 4 47 27 PM copy](https://user-images.githubusercontent.com/2775448/169626008-08d2ee77-f059-49e3-9f00-80cbdbaf6ec0.png)

- Notification Pane:
![Screen Shot 2022-05-20 at 4 51 22 PM](https://user-images.githubusercontent.com/2775448/169626028-2f766c99-3dbe-4d8f-91e5-0634248450b9.png)

- Notification dialog:
![Screen Shot 2022-05-20 at 4 51 28 PM](https://user-images.githubusercontent.com/2775448/169626042-d75958d1-8db1-470f-859a-ed610dea3b98.png)

# individual_merge_blocked:
- Notification:
![Screen Shot 2022-05-20 at 4 47 27 PM](https://user-images.githubusercontent.com/2775448/169626096-de282367-b4d0-4aa6-911b-4c8bcd67eca1.png)

- Notification Pane:
![Screen Shot 2022-05-20 at 4 50 32 PM](https://user-images.githubusercontent.com/2775448/169626117-caeeaaba-71ba-45e7-8992-91d3fab5b190.png)

- Notification Dialog:

![Screen Shot 2022-05-20 at 5 05 44 PM](https://user-images.githubusercontent.com/2775448/169626169-a57099a1-e9c2-4971-9242-9fed5516c14d.png)


QA Notes:
It will be important to check for regression with other notifications, most notably individual_merge_complete (i.e., "clicking "Allow" for a merge request should still behave as it did previously).
Other than that, just make sure that merge requests can be made with the appropriate notifications in all three locations (the dialog, the notifications page proper, and the dialog the pulls up when you click on the notification list item therein).
Similarly make sure that you see the blocked merge notification in all three spots as pictured above.
